doc/decisions/4_in_review/change_tracking.md

```diff
@@ -174,13 +174,26 @@ The changetracking plugin needs to export at least functions for the following t
 
 ## Decision
 
+Plugins and application developer can enable changetracking via a contract.
+Store deep-duped returned keys in a separate KeySet, which we might be able to reuse as internal cache later on.
+The whole changetracking logic lives within `libelektra-kdb`.
+We provide an API for developers with `libelektra-kdb`.
+
 ## Rationale
 
+This decision meets all the constraints.
+
+We are not altering the behaviour of already existing functions, so everything is transparent for applications using the public Elektra API.
+As changetracking is an opt-in feature, it will create zero overhead in usecases where it is not needed.
```

Comment:
This contradicts with the Rationale below. (`struct _KDB` gets an additional pointer etc.).

```suggestion
As changetracking is an opt-in feature, it will create negligible overhead if not needed.
```

Reply:
I think "overhead" refers to runtime here, so how about:

```suggestion
As changetracking is an opt-in feature, it will create zero runtime overhead and negligible memory overhead in usecases where it is not needed.
```

Reply:
What would be the difference between runtime and memory overhead?

Reply:
Oh you mean CPU cycles?